runtime/errors.go

```diff
@@ -10,6 +10,9 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+// ErrorHandlerFunc handles the error as a gRPC error generated via status package and replies to the request.
+type ErrorHandlerFunc func(context.Context, *ServeMux, Marshaler, http.ResponseWriter, *http.Request, error)
```

Comment:
A single signature for error handling is great, but to support streaming errors, this would need more info -- mainly it needs to know whether or not the HTTP status and headers have already been sent. That way the handler can choose to write to the body in this case.